### PR TITLE
Fix bug with implementation of `bigint.pow()` when the exponent is negative

### LINKED ISSUE #17704: Fix bug with `bigint.pow` in BigInteger module when exponent is negative
When the exponent is a positive integer, we use the uint version, but when the exponent is negative we send the base as the mod value to work around the fact that GMP doesn't provide a pow version for integers and this wouldn't yield the same result if `y == -1` or if the base is negative.  It looks like this function was added along with a bunch of other int versions - I don't see a review and the PR did not add tests of this function specifically.

### PR DESCRIPTION
**Pull request by lydia-duncan**
The original implementation was calling `bigint.powm()` under the covers when
the exponent was negative, sending the `base` argument again as the `mod`
argument - however, through testing via Wolfram Alpha there were many cases
where sending the `base` as the `mod` value would result in a different result,
and actually calling the `bigint.mod` method in these cases would result in
segfaults.  Instead, since a negative exponent results in a fractional value for
all bases except 1 and -1, return 0 for most bases and return the base itself
when the base is 1.  When the base is -1, return -1 when the exponent is odd and
1 when the exponent is even.

Resolves #17704 

Add a test of negative exponents with `bigint.pow()`.  This test segfaulted prior
to this change.  It now passes.

Passed a full paratest with futures

### CODE REVIEW

File: test/library/standard/GMP/powCheck.good [hunk 1]
```diff
@@ -0,0 +1,8 @@
+2^-1 = 0
+2^-2 = 0
+4^-1 = 0
+4^-2 = 0
+1^-1 = 1
+1^-2 = 1
+-1^-1 = -1
+-1^-2 = -1
```
Comment (bradcray) [inline test/library/standard/GMP/powCheck.good:8]: Using `bc`, I get:

> -1^-2
> 1

Comment (lydia-duncan) [reply test/library/standard/GMP/powCheck.good:8]: Oops, looks like I needed parenthesis.  I'll fix that

Comment (lydia-duncan) [reply test/library/standard/GMP/powCheck.good]: Okay, fixed.  Have another look?

**Comment by bradcray** [comment]
Reviewing open PRs:  Is this post-release at this point, or are we considering it for 1.25?

**Comment by lydia-duncan** [comment]
Nobody volunteered to review it yesterday, and since it's a code change I think that probably means it shouldn't go in today?  It'd be nice to get it in, but I wouldn't lose sleep over it waiting until next release

**Comment by bradcray** [comment]
I think that sounds right, thanks.

File: modules/standard/BigInteger.chpl [hunk 2]
```diff
@@ -2677,24 +2677,35 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
...
-        const exp_ = new bigint(exp);
-
-        mpz_powm(this.mpz, base.mpz, exp_.mpz, base.mpz);
+        if (base == 1 || (base == -1 && Math.abs(exp) % 2 == 1)) {
+          this = base;
+        } else if (base == -1 && Math.abs(exp) % 2 == 0) {
+          this = 1;
+        } else {
+          this = 0;
+        }
```
Comment (mppf) [inline modules/standard/BigInteger.chpl:2686]: Can you put this code into a helper function please?

**Review by mppf — APPROVED** [review]

**Pull request merged**